code/game/objects/machinery/OpTable.dm

```diff
@@ -230,3 +227,37 @@
 		return 0
 
 	return 1
+
+/obj/machinery/optable/update_icon_state() //This is to simplify and clean some above code.
+	. = ..()
+	if(!victim)
+		icon_state = initial(icon_state)
+		return
+	if(victim.handle_pulse())
+		icon_state = initial(icon_state) + "-active"
```

Comment:
You missed an early return it seems. Also this could be made in one if return, not two

Reply:
if (victim?handle_pulse())
    icon = thing
    return
icon = other thing

Reply:
And anyway, this doesn't work.